src/MQTTAsync.c

```diff
@@ -2552,7 +2552,8 @@ static thread_return_type WINAPI MQTTAsync_receiveThread(void* n)
 		if (rc == SOCKET_ERROR)
 		{
 			Log(TRACE_MINIMUM, -1, "Error from MQTTAsync_cycle() - removing socket %d", sock);
-			nextOrClose(m, rc, "socket error");
+			//nextOrClose(m, rc, "socket error");
+			MQTTAsync_disconnect_internal(m, 0);
```

Comment:
@icraggs I'm not sure what the nextOrClose method does. Can you please explain? My expectation is that after an ssl error the client gets disconnected. This is also how it works for ssl errors that occur during normal publish.
https://github.com/eclipse/paho.mqtt.c/blob/master/src/MQTTAsync.c#L1996

Reply:
nextOrClose is called when a connection fails, to take the next action.  Closing the current connection, and then reconnecting with a new connection if that is needed.  A new connection is needed if MQTTVERSION_DEFAULT is set as that reconnects with MQTT 3.1, or automatic reconnect is set - so we kick off the automatic reconnect processing.

Reply:
I saw you recently added disconnect_internal in 275843846da2f5a0e4e27586f1e6fbd2f9e2f67b